src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.Special.cs

```diff
@@ -134,9 +134,31 @@ public CborSpecialValue ReadSpecialValue()
             }
         }
 
-        private static float Read16BitFloatBigEndian(ReadOnlySpan<byte> buffer)
+        private const float foo = 5.9604644775390625e-08f; // 2 ^ -24
+
+        // half-precision float decoder adapted from https://tools.ietf.org/html/rfc7049#appendix-D
+        private static double ReadHalfBigEndian(ReadOnlySpan<byte> buffer)
         {
-            throw new NotImplementedException(nameof(Read16BitFloatBigEndian));
+            int half = (buffer[0] << 8) + buffer[1];
+            bool isNegative = (half >> 15) != 0;
+            int exp = (half >> 10) & 0x1f;
+            int mant = half & 0x3ff;
+            double value;
+
+            if (exp == 0)
+            {
+                value = mant * 5.9604644775390625e-08 /* precomputed 2^-24 */;
```

Comment:
ideally the JIT would understand Math.Pow and replace it with a constant. @tannergooding ?

Reply:
This is blocked by the more general issue that not all platforms compute particular results consistently which requires additional work to either port the implementation to managed code or find a stable/good/performant C implementation that can be shared.